### PR TITLE
【WIP】koordlet: perform operations only under the specified cpuburst strategy and improve config logging

### PR DESCRIPTION
**Pull request by yangfeiyu20102011**
image
Under the specified cpuBurstOnly strategy, the plugin cpuburst still periodically updates the CFS values, which is not expected. Determine the corresponding cpuburst strategy at the beginning and then perform the appropriate actions.
image

### V. Checklist

- [ ] I have written necessary docs and comments
- [ ] I have added necessary unit tests and integration tests
- [ ] All checks passed in `make test`

### CODE REVIEW

File: pkg/koordlet/qosmanager/plugins/cpuburst/cpu_burst.go [hunk 1]
```diff
@@ -241,15 +241,21 @@ func (b *cpuBurst) start() {
...
 			continue
 		}
-		klog.V(5).Infof("get pod %v/%v cpu burst config: %v", podMeta.Pod.Namespace, podMeta.Pod.Name, cpuBurstCfg)
-		// set cpu.cfs_burst_us for pod and containers
-		b.applyCPUBurst(cpuBurstCfg, podMeta)
-		// scale cpu.cfs_quota_us for pod and containers
-		b.applyCFSQuotaBurst(cpuBurstCfg, podMeta, nodeState)
+		klog.V(5).Infof("get pod %v/%v cpu burst config: %#v", podMeta.Pod.Namespace, podMeta.Pod.Name, cpuBurstCfg)
+
+		if cpuBurstEnabled(cpuBurstCfg.Policy) {
```
Comment (saintube) [inline pkg/koordlet/qosmanager/plugins/cpuburst/cpu_burst.go:250]: please note that the strategy should support reseting the related cgroups when the policy disables

Comment (yangfeiyu20102011) [reply pkg/koordlet/qosmanager/plugins/cpuburst/cpu_burst.go:250]: > please note that the strategy should support reseting the related cgroups when the policy disables

When both CPUNormalization and CPUBurst are enabled, CFS will be repeatedly changed by the plugins.
How about adding a flag variable, when `CPUBurstPolicy` changed，call `b.applyCPUBurst(cpuBurstCfg, podMeta)` and `b.applyCFSQuotaBurst(cpuBurstCfg, podMeta, nodeState)` as before. @saintube 

```
func (b *cpuBurst) start(){
...
var flag bool
if nodeSLO.Spec.CPUBurstStrategy changed {
flag = true
}
...
if cpuBurstEnabled(cpuBurstCfg.Policy) || flag {
			// set cpu.cfs_burst_us for pod and containers
			b.applyCPUBurst(cpuBurstCfg, podMeta)
}

if cfsQuotaBurstEnabled(cpuBurstCfg.Policy) || flag {
			// scale cpu.cfs_quota_us for pod and containers
			b.applyCFSQuotaBurst(cpuBurstCfg, podMeta, nodeState)
}

}
```

Comment (saintube) [reply pkg/koordlet/qosmanager/plugins/cpuburst/cpu_burst.go:250]: Maybe we can add a new policy `"ignored"` to describe the behavior that stops the cgroups reconciliation, so the existing policies are still backward-compatible.

Comment (yangfeiyu20102011) [reply pkg/koordlet/qosmanager/plugins/cpuburst/cpu_burst.go:250]: > Maybe we can add a new policy `"ignored"` to describe the behavior that stops the cgroups reconciliation, so the existing policies are still backward-compatible.

But someone may still want to use `cpuBurstOnly`,  `applyCPUBurst()` is also needed. @saintube

Comment (saintube) [reply pkg/koordlet/qosmanager/plugins/cpuburst/cpu_burst.go:250]: Yes. The user may switch the policy from `auto` to `cpuBurstOnly`. He/She should reset the cfs_quota. 
Adding a separate attribute is a good idea not to break the backward compatibility of the current policies. Possibly, we can define a new field in the CPUBurstStrategy to adjust the configurations dynamically.

**Pull request closed without merging**